### DISCUSSION TITLE
Can someone explain the purpose of specifying the webroot?

### DISCUSSION THREAD

**Discussion by benwiley4000** [acmesh-official/acme.sh#4404] [Q&A]
My app that will be served is running from a docker container and I will be mounting the certs as a volume. So it's a bit complicated to specify the webroot on disk. Can someone explain the function of specifying the webroot? Can I just create an empty directory? Or does it need to contain the actual web content that will be served when the URL is hit?

**Comment by chandave** [comment]
"--webroot" defines the directory where acme.sh should place a text file containing the challenge token used by the HTTP-01 verification process.  Actually, it will be a directory_structure+text file (.well_known/acme-challenge/_token_file_name_). The challenge token is provided by the ACME server. acme.sh handles grabbing the challenge token from the ACME server and writing it into your defined directory.

That text file must be served by your web server when the ACME challenge server makes an HTTP GET request for that challenge file (URL: http://your_server/.well_known/acme-challenge/token_file_name).

> Can I just create an empty directory? Or does it need to contain the actual web content that will be served when the URL is hit?

That directory has to exist when acme.sh executes and must also be served by your web server. It does not need to have any web content.
